### PR TITLE
fix: always call `WandbLogger.experiment` first in `_call_setup_hook` to ensure tensorboard logs sync to wandb

### PR DESCRIPTION
**Pull request by ringohoffman**
## What does this PR do?

In `lightning.pytorch.trainer.call._call_setup_hook`, sort the logger list when calling `experiment` on all loggers for the first time so that `WandbLogger.experiment` is always called before `TensorboardLogger.experiment`.

This ensures that `wandb.init()` is called before any tensorboard `SummaryWriter` are created, thereby guaranteeing that tensorboard logs can be synchronized to Weights and Biases.

See:

* https://github.com/wandb/wandb/issues/1782#issuecomment-779161203

Also [from the Weights and Biases tensorboard integrations page](https://docs.wandb.ai/guides/integrations/tensorboard/):

> You must call either `wandb.init` or `wandb.tensorboard.patch` before calling `tf.summary.create_file_writer` or constructing a `SummaryWriter` via `torch.utils.tensorboard`.

Fixes N/A

<details>
  <summary><b>Before submitting</b></summary>

- [ ] Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc: @Borda @awaelchli

### CODE REVIEW

**Comment by ringohoffman** [comment]
cc: @Borda

**Comment by ringohoffman** [comment]
Thanks for approving! Is this able to merge?
